spec/unit/provider/user/aix_spec.rb

```diff
@@ -70,11 +70,23 @@
     end
   end
 
-  describe '.groups_to_groups' do
+  describe '.groups_attribute_to_property' do
+    it "reads the user's groups from the etc/groups file" do
+      groups = ['system', 'adm']
+      Puppet::Util::POSIX.expects(:groups_of).with(resource[:name]).returns(groups)
```

Comment:
I think in these scenarios we should just stub the method rather than have it expect specific inputs. In this unit test this part would be testing implementation, rather than strictly functionality of the method under test.

This is something that I'm learning to do as well, and I think it makes more sense to unit test that way.